pkg/selectors/kernel.go

```diff
@@ -411,88 +411,131 @@ func argSelectorType(arg *v1alpha1.ArgSelector, sig []v1alpha1.KProbeArg) (uint3
 	return 0, fmt.Errorf("argFilter for unknown index")
 }
 
+func writePortRangeInMap(v string, ty uint32, op uint32, m *ValueMap) error {
+	// We store the start and end of the range as uint64s for unsigned values, and as int64s
+	// for signed values. This is to allow both a signed range from -5 to 5, and also an
+	// unsigned range where at least limit exceeds max(int64). If we only stored the range
+	// limits as uint64s, then the range from -5 to 5 would be interpretted as being from
+	// 5 to uint64(-5), which is the literal opposite of what was intended. If we only stored
+	// the range as int64s, then we couldn't correctly accommodate values that exceed max(int64),
+	// for similar reasons.
+	var uRangeVal [2]uint64
```

Comment:
If this function is only handling port ranges (and not ranges of anything else) then it could all be switched to unsigned values for simplicity.